test/integration/gardenlet/seed/seed/seed_test.go

```diff
@@ -333,12 +369,18 @@ var _ = Describe("Seed controller tests", func() {
 					By("Delete Seed")
 					Expect(testClient.Delete(ctx, seed)).To(Succeed())
 
-					By("Verify that the seed system components have been deleted")
-					Eventually(func(g Gomega) []resourcesv1alpha1.ManagedResource {
-						managedResourceList := &resourcesv1alpha1.ManagedResourceList{}
-						g.Expect(testClient.List(ctx, managedResourceList, client.InNamespace(testNamespace.Name))).To(Succeed())
-						return managedResourceList.Items
-					}).Should(BeEmpty())
+					if seedIsGarden {
+						// The CRDs are cleaned up by the Destroy function of GRM, in case the seed is not garden. So It might happen that, before we fetch the
+						// ManagedResourceList is empty, the CRDs are already gone. Since the gardener-resource-manager is deleted only after all the managedresources
+						// are gone, we don't need to assert it seperately. In case the seed is garden, the Destroy is called by the gardener-operator and since it's
```

Comment:
```suggestion
						// are gone, we don't need to assert it separately. In case the seed is garden, the Destroy is called by the gardener-operator and since it's
```